barretenberg/cpp/src/barretenberg/eccvm/msm_builder.hpp

```diff
@@ -43,138 +45,138 @@ class ECCVMMSMMBuilder {
+        //      | 0 | 1 |        | Block_{0}                      | <-- pc = total_number_of_muls
+        //      | - | - |        | Block_{1}                      | <-- pc = total_number_of_muls-(num muls in msm 0)
+        //    1 | # | # | -1     |   ...                          | ...
+        //    3 | # | # | -3     | Block_{total_number_of_muls-1} | <-- pc = num muls in last msm
+        //    5 | # | # | -5
+        //    7 | # | # | -7
+        //    9 | # | # | -9
+        //   11 | # | # | -11
+        //   13 | # | # | -13
+        //   15 | # | # | -15
+
+        const size_t num_rows_in_read_counts_table =
+            static_cast<size_t>(total_number_of_muls) * (eccvm::POINT_TABLE_SIZE >> 1);
+        std::array<std::vector<size_t>, 2> point_table_read_counts;
+        point_table_read_counts[0].reserve(num_rows_in_read_counts_table);
+        point_table_read_counts[1].reserve(num_rows_in_read_counts_table);
+        for (size_t i = 0; i < num_rows_in_read_counts_table; ++i) {
             point_table_read_counts[0].emplace_back(0);
             point_table_read_counts[1].emplace_back(0);
         }
-        const auto update_read_counts = [&](const size_t pc, const int slice) {
-            // When we compute our wnaf/point tables, we start with the point with the largest pc value.
-            // i.e. if we are reading a slice for point with a point counter value `pc`,
-            // its position in the wnaf/point table (relative to other points) will be `total_number_of_muls - pc`
-            const size_t pc_delta = total_number_of_muls - pc;
-            const size_t pc_offset = pc_delta * 8;
-            bool slice_negative = slice < 0;
-            const int slice_row = (slice + 15) / 2;
-
-            const size_t column_index = slice_negative ? 1 : 0;
 
+        const auto update_read_count = [&point_table_read_counts](const size_t point_idx, const int slice) {
             /**
-             * When computing `point_table_read_counts`, we need the *table index* that a given point belongs to.
-             * the slice value is in *compressed* windowed-non-adjacent-form format:
-             * A non-compressed WNAF slice is in the range: `-15, -13, ..., 15`
-             * In compressed form, tney become `0, ..., 15`
+             * The wNAF digits lie in the range -15, -13, ..., 13, 15.
              * The *point table* format is the following:
-             * (for positive point table) T[0] = P, T[1] = PT, ..., T[7] = 15P
+             * (for positive point table) T[0] =  P, T[1] =  3P, ..., T[7]  =  15P
              * (for negative point table) T[0] = -P, T[1] = -3P, ..., T[15] = -15P
              * i.e. if the slice value is negative, we can use the compressed WNAF directly as the table index
-             *      if the slice value is positive, we must take `15 - compressedWNAF` to get the table index
+             *      if the slice value is positive, we must take 15 - (compressed wNAF) to get the table index
```

Comment:
it'd be great to have a sentence explaining why the positive-negative separation is required

Reply:
It's not required, idk why he did it this way.